File: 1615-range-sum-of-sorted-subarray-sums/range-sum-of-sorted-subarray-sums.cpp
class Solution {
public:
    int MOD = 1e9 + 7;
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        vector<int> pfxSum;
        for (int i = 0; i < n; ++i) {
            int pSum = 0;
            for (int j = i; j < n; ++j) {
                pSum += nums[j];
                pfxSum.push_back(pSum);
            }
        }
        sort(pfxSum.begin(), pfxSum.end());
        int ans = 0;
        for (int i = left - 1; i <= right - 1; ++i) {
            ans = (ans + pfxSum[i]) % MOD;
        }
        return ans;
    }
};

File: 1615-range-sum-of-sorted-subarray-sums/range-sum-of-sorted-subarray-sums.java
class Solution {
    public int MOD = 1000000007;

    public int rangeSum(int[] nums, int n, int left, int right) {
        ArrayList<Integer> pfxSum = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            int sum = 0;
            for (int j = i; j < n; j++) {
                sum += nums[j];
                pfxSum.add(sum);
            }
        }
        Collections.sort(pfxSum);
        int ans = 0;
        for (int i = left - 1; i < right; ++i) {
            ans = (ans + pfxSum.get(i)) % MOD;
        }
        return ans;
    }
}